### PR TITLE
Status update comments

### LINKED ISSUE #723: Status update comments
Move Hire / Reject into comments and bring status comments

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Feature

## Related issues
Resolve #723

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/e1535b0c-43d0-4e1a-8072-323eb2bf4ccd

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/a6c12afb-159e-43e5-8d30-fb330f0444b6

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/hire/assets/81121562/5b6786c5-18c6-4f4e-b65c-1dd99ed3b82a

File: prisma/schema.prisma [hunk 1]
```diff
@@ -483,6 +483,8 @@ model Comment {
 
   reactions Reaction[]
 
+  status InterviewStatus? @default(NEW)
```
Comment (nyet-ty) [inline prisma/schema.prisma:486]: do we need default status if it optional for comment?

**Review by nyet-ty — CHANGES_REQUESTED** [review]

File: src/components/CommentCreateForm/CommentCreateForm.tsx [hunk 2]
```diff
@@ -14,6 +18,10 @@ interface CommentCreateFormProps extends Omit<React.ComponentProps<typeof Commen
     onChange?: (comment: CommentSchema) => void;
     attachFormatter?: ComponentProps<typeof FormEditor>['attachFormatter'];
     uploadLink?: string;
+    rejectReasons?: RejectReason[];
+    statusDropdown?: React.ReactNode;
```
Comment (tsumo) [inline src/components/CommentCreateForm/CommentCreateForm.tsx:22]: Unused prop?

File: src/components/CommentCreateForm/CommentCreateForm.tsx [hunk 3]
```diff
@@ -47,24 +60,58 @@ const CommentCreateForm: React.FC<CommentCreateFormProps> = ({
...
+                    setText(rejectReasons[0]?.text);
+                    setIsOpen(false);
+                    setStatusInterview(InterviewStatus.REJECTED);
+                },
+                text: tr('Reject'),
+            },
+        );
+
+        return items;
+    }, [rejectReasons]);
```
Comment (tsumo) [inline src/components/CommentCreateForm/CommentCreateForm.tsx:113]: Missing dependancy

File: src/components/CommentCreateForm/CommentCreateForm.tsx [hunk 4]
```diff
@@ -47,24 +60,58 @@ const CommentCreateForm: React.FC<CommentCreateFormProps> = ({
...
         setBusy(false);
         setFocused(false);
         setText('');
+        setStatusInterview(undefined);
+
         onCancel?.();
     }, [onCancel]);
 
+    const statusInterviewMenuItems = useMemo<DropdownMenuItem[]>(() => {
+        const items: DropdownMenuItem[] = [];
```
Comment (tsumo) [inline src/components/CommentCreateForm/CommentCreateForm.tsx:89]: Just return array directly here `return [ {}, {}, ]`

File: src/components/CommentCreateForm/CommentCreateForm.tsx [hunk 5]
```diff
@@ -73,13 +120,66 @@ const CommentCreateForm: React.FC<CommentCreateFormProps> = ({
                 text={text}
                 focused={focused}
                 busy={busy}
+                status={statusInterview}
                 onChange={onCommentChange}
                 onSubmit={onCommentSubmit}
                 onCancel={onCancelCreate}
                 onFocus={onCommentFocus}
+                childrenProps={
+                    visibleRejectOption && (
```
Comment (tsumo) [inline src/components/CommentCreateForm/CommentCreateForm.tsx:129]: nullable

File: src/components/CommentForm/CommentForm.tsx [hunk 6]
```diff
@@ -13,10 +14,12 @@ import s from './CommentForm.module.css';
 
 interface CommentFormProps {
     actionButton: React.ReactNode;
+    childrenProps?: React.ReactNode;
```
Comment (tsumo) [inline src/components/CommentForm/CommentForm.tsx:17]: Looks like you dont need this prop, you can move RejectInterviewStatus into this component

File: src/components/CommentForm/CommentForm.module.css [hunk 7]
```diff
@@ -33,6 +33,12 @@
     left: -7px;
 }
 
+.CommentFormRejectInterviewStatusWrapper {
```
Comment (tsumo) [inline src/components/CommentForm/CommentForm.module.css:36]: Does this css really do anything?

File: src/components/CommentView/CommentView.i18n/en.json [hunk 8]
```diff
@@ -1,5 +1,6 @@
 {
     "Edit": "",
     "Delete": "",
-    "Save": ""
+    "Save": "",
+    "Status interview: ": ""
```
Comment (tsumo) [inline src/components/CommentView/CommentView.i18n/en.json:5]: Interview status

File: src/components/CommentView/CommentView.tsx [hunk 9]
```diff
@@ -149,34 +174,79 @@ export const CommentView: FC<CommentViewProps> = ({
...
-                                        onClick={item.onClick}
-                                    >
-                                        {item.label}
-                                    </MenuItem>
-                                )}
-                            />
-                        </div>
-                    </CardInfo>
+                    {status === InterviewStatus.REJECTED || status === InterviewStatus.HIRED ? (
+                        <CardInfo className={cn(s.CardInfo, InterviewStatusColors[interviewStatusChip])}>
```
Comment (tsumo) [inline src/components/CommentView/CommentView.tsx:178]: looks heavy, maybe move this to a new component InterviewStatusCommentCard?

File: src/components/RejectInterviewStatus/RejectInterviewStatus.tsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+import s from './RejectInterviewStatus.module.css';
+import { tr } from './RejectInterviewStatus.i18n';
+
+interface RejectInterviewStatusProps {
+    rejectReasons: RejectReason[];
+    onChangeRejectReasons?: (selected?: any) => void;
+}
+
+export const RejectInterviewStatus = ({
```
Comment (tsumo) [inline src/components/RejectInterviewStatus/RejectInterviewStatus.tsx:16]: InterviewRejectReasonDropdown

File: src/trpc/routers/commentRouter.ts [hunk 11]
```diff
@@ -1,13 +1,60 @@
...
+import { historyEventMethods } from '../../modules/historyEventMethods';
 
 export const commentRouter = router({
     create: protectedProcedure
         .input(createCommentSchema)
         .use(accessMiddlewares.comment.create)
-        .mutation(({ input }) => {
+        .mutation(async ({ input, ctx }) => {
+            if ('interviewId' in input.target) {
+                const interview = await interviewMethods.getById(input.target.interviewId);
```
Comment (tsumo) [inline src/trpc/routers/commentRouter.ts:19]: don't need this, get interviewWithSections(input.target.interviewId)

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Review by nyet-ty — APPROVED** [review]

**Pull request merged**